configure settings.xml during build to define USB not supported

AlwinEsch opened the pull request:
This change it to settings.xml.in and set on add-on compile the settings usable or not. Is done by change the `<visible>` value and set of default connection type. The settings are still inside to prevent problems if something tries somewhere to read it.

Further can for build the cmake configuration value "-DUSB_DEVICE_SUPPORT=0" used and if wanted to disable USB support.

XML with USB support disabled
```xml
        <!-- USB device not suppoted on this add-on, due to this the following setting is not usable and marked as not visible. -->
        <setting id="device_connection" type="integer" label="30100" help="30500">
          <level>0</level>
          <visible>false</visible>
          <default>1</default>
          <constraints>
            <options>
              <option label="30200">0</option>
              <option label="30201">1</option>
            </options>
          </constraints>
          <control type="spinner" format="integer"/>
        </setting>

        <!-- USB device not suppoted on this add-on, due to this the following setting is not usable and marked as not visible. -->
        <setting id="device_connection_usb_index" parent="device_connection" type="integer" label="30101" help="30501">
          <level>0</level>
          <visible>false</visible>
          <default>0</default>
          <constraints>
            <minimum>0</minimum>
            <step>1</step>
            <maximum>8</maximum>
          </constraints>
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">0</dependency>
          </dependencies>
          <control type="spinner" format="integer"/>
        </setting>

        <setting id="device_connection_tcp_host" parent="device_connection" type="string" label="30102" help="30502">
          <level>0</level>
          <default/>
          <constraints>
            <allowempty>true</allowempty>
          </constraints>
          <!--
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">1</dependency>
          </dependencies>
          -->
          <control type="edit" format="ip">
            <heading>30102</heading>
          </control>
        </setting>

        <setting id="device_connection_tcp_port" parent="device_connection" type="integer" label="30103" help="30503">
          <level>0</level>
          <default>1234</default>
          <constraints>
            <allowempty>false</allowempty>
            <minimum>0</minimum>
            <maximum>65535</maximum>
          </constraints>
          <!--
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">1</dependency>
          </dependencies>
          -->
          <control type="edit" format="integer">
            <heading>30103</heading>
          </control>
        </setting>
```

XML with USB support enabled
```xml
        <!-- USB device supported on this add-on. -->
        <setting id="device_connection" type="integer" label="30100" help="30500">
          <level>0</level>
          <visible>true</visible>
          <default>0</default>
          <constraints>
            <options>
              <option label="30200">0</option>
              <option label="30201">1</option>
            </options>
          </constraints>
          <control type="spinner" format="integer"/>
        </setting>

        <!-- USB device supported on this add-on. -->
        <setting id="device_connection_usb_index" parent="device_connection" type="integer" label="30101" help="30501">
          <level>0</level>
          <visible>true</visible>
          <default>0</default>
          <constraints>
            <minimum>0</minimum>
            <step>1</step>
            <maximum>8</maximum>
          </constraints>
          <dependencies>
            <dependency type="visible" setting="device_connection" operator="is">0</dependency>
          </dependencies>
          <control type="spinner" format="integer"/>
        </setting>
```

AlwinEsch commented on pvr.rtlradio/resources/settings.xml.in:
```diff
@@ -53,9 +59,11 @@
           <dependencies>
             <dependency type="visible" setting="device_connection" operator="is">1</dependency>
           </dependencies>
+          @SETTINGS_XML_IN_NO_USB_END@
```
I just added this as an afterthought. Found the case that USB was previously activated during the test and this was no longer displayed.

AlwinEsch commented on src/addon.cpp:
```diff
@@ -1153,8 +1153,12 @@ ADDON_STATUS addon::Create(void)
           kodi::addon::GetSettingEnum("device_connection", device_connection::usb);
+#else
+      m_settings.device_connection = device_connection::rtltcp;
+#endif
```
Because USB was enabled, it is now forced to use tcp.

AlwinEsch commented:
Maybe another settings text could be good🤔?
Using “- rtl_sdr” as the start looks kind of weird.
![Screenshot 2024-04-03 164704](https://github.com/kodi-pvr/pvr.rtlradio/assets/6879739/e204ac33-8703-4a86-9eef-3862f59676dd)

AlwinEsch commented:
Have added second commit to add groups and to remove the "- ", further renamed with remove of "rtl_tcp " in text.
![Screenshot 2024-04-03 171850](https://github.com/kodi-pvr/pvr.rtlradio/assets/6879739/b6e1f498-6fff-41d8-bc12-35181255a2ea)
![Screenshot 2024-04-03 172054](https://github.com/kodi-pvr/pvr.rtlradio/assets/6879739/446b15bc-736d-4dd8-ae80-68b4657b29b7)
![Screenshot 2024-04-03 172239](https://github.com/kodi-pvr/pvr.rtlradio/assets/6879739/7bb7a581-ad41-4e6b-a58e-93398eac67ff)

AlwinEsch committed:
set IP 127.0.0.1 as default in add-on setting

ksooo commented on CMakeLists.txt:
```diff
@@ -4,11 +4,25 @@ project(pvr.rtlradio)
+
+# Set values used about configure of settings.xml.in on add-on
+if(NOT USB_DEVICE_SUPPORT)
+  set(SETTINGS_XML_IN_CONNECTION_TYPE_NOTE "USB device not suppoted on this add-on, due to this the following setting is not usable and marked as not visible.")
```
Typo: suppoted -> supported

AlwinEsch replied:
Thanks, always somewhere a typo 😅.

ksooo commented on src/addon.cpp:
```diff
@@ -1153,8 +1153,12 @@ ADDON_STATUS addon::Create(void)
+#else
+      m_settings.device_connection = device_connection::rtltcp;
+#endif
       m_settings.device_connection_usb_index =
           kodi::addon::GetSettingInt("device_connection_usb_index", 0);
```
You only need to read this property value if `USB_DEVICE_SUPPORT`is defined.

AlwinEsch replied:
Have changed there, also on the SettingsChange function is now the `USB_DEVICE_SUPPORT`

AlwinEsch committed:
configure settings.xml during build to define USB not supported

This change it to settings.xml.in and set on add-on compile the settings usable or not.
Is done by change the `<visible>` value and set of default connection type.
The settings are still inside to prevent problems if something tries somewhere to read it.

Further can for build the cmake configuration value "-DUSB_DEVICE_SUPPORT=0" used and if wanted to disable USB support.

AlwinEsch committed:
fix settings.xml entry hide in condition type disabled and region changed

ksooo reviewed (changes requested):
addon.cpp does not compile if `USB_DEVICE_SUPPORT` is not defined. Look at line 1440, there is an 'else if' without an 'if' then.

ksooo commented:
Shouldn't `Jenkinsfile` be adjusted to build all platforms again?

AlwinEsch committed:
change settings to use groups and change strings to not show "- "

AlwinEsch committed:
use all platforms in Jenkins

AlwinEsch commented:
> Shouldn't `Jenkinsfile` be adjusted to build all platforms again?

Yes, now all OS works, have added a commit about.

AlwinEsch commented:
> addon.cpp does not compile if `USB_DEVICE_SUPPORT` is not defined. Look at line 1440, there is an 'else if' without an 'if' then.

Ups, is fixed now. But Jenkins brings now the next and iOS/tvOS become USB device support removed too.
It brings now:
```
In file included from os/darwin_usb.c:50:

os/darwin_usb.h:28:10: fatal error: 'IOKit/IOTypes.h' file not found

#include <IOKit/IOTypes.h>
```
Seems something updated in Jenkins on the morning.

AlwinEsch commented:
On new added commit becomes the USB support on Android removed too and Jenkins now happy again on all OS.
EDIT: Happy on all OS after several additional commits.

AlwinEsch committed:
dont use USB device support in iOS, tvOS and Android devices

AlwinEsch committed:
don't build libout123 in depends mpg123

This to fix build errors on Darwin Embedded system (iOS & tvOS).

AlwinEsch committed:
don't use "EGL/egl.h" within whole Darwin system

This fix new Darwin Embedded error:
```
/Users/Shared/jenkins/workspace/binary-addons/kodi-tvos-aarch64-Omega/tools/depends/target/binary-addons/pvr.rtlradio/src/gui/channelsettings.cpp:38:10: fatal error: 'EGL/egl.h' file not found
         ^~~~~~~~~~~
32 warnings and 1 error generated.
```

ksooo reviewed (approved):
LGTM
